drivers/python/rethinkdb/_export.py

```diff
@@ -188,12 +205,21 @@ def write_table_metadata(progress, conn, db, table, base_path):
 # This is called through rdb_call_wrapper and may be called multiple times if
 # connection errors occur.  In order to facilitate this, we do an order_by by the
 # primary key so that we only ever get a given row once.
-def read_table_into_queue(progress, conn, db, table, pkey, task_queue, progress_info, exit_event):
+def read_table_into_queue(progress, conn, db, table, pkey, keys, index, task_queue, progress_info, exit_event):
     read_rows = 0
-    if progress[0] is None:
-        cursor = r.db(db).table(table).order_by(index=pkey).run(conn, time_format="raw", binary_format='raw')
+    table = r.db(db).table(table)
+
+    if progress[0] is not None:
+        table = table.between(progress[0], None, left_bound="open")
+
+    if keys is None:
+        table = table.order_by(index=pkey)
+    elif index is not None:
+        table = table.get_all(*keys, index=index).order_by(pkey)
```

Comment:
Sorry, guys, lost track of this.  It should be possible to get this to work, but it requires 6 different queries based on whether `keys` or `index` are set, and whether we're resuming after a connection interruption.  I would like to merge this functionality into `next`, but it requires some careful handling of the queries so I'd much rather take care of it myself.

Would it be alright if I branched this and merged it in separately?